dd-sdk-android-ndk/src/test/cpp/test_datetime_utils.cpp

```diff
@@ -0,0 +1,17 @@
+#include <regex>
+
+#include "greatest/greatest.h"
+#include "utils/datetime.h"
+
+TEST test_generate_event_date_format(void) {
+    char buffer[100];
+    format_date(buffer, sizeof(buffer) / sizeof(buffer[0]), "%Y-%m-%d'T'%H:%M:%S.000Z");
+            ASSERT(std::regex_match(buffer, std::regex(
+            "[0-9]{4}-[0-9]{2}-[0-9]{2}'T'[0-9]{2}:[0-9]{2}:[0-9]{2}.000Z")));
+            PASS();
```

Comment:
This test is incomplete I think. It'd be best to ensure that the string has the right format, and the right value.

Reply:
Well problem is that I really cannot assert the value as it is the current time so it will differ in number of nanos.

Reply:
Why would it be an issue to spend more time in test than on the implementation ? Happens to me all the time. 
Test and code should grow hand in hand. You could return a fixed value for every date and still have all your tests green.